whylog/tests/tests_teacher/test_rule_validation.py

```diff
@@ -1,24 +1,37 @@
 from whylog.config.filename_matchers import WildCardFilenameMatcher
 from whylog.config.log_type import LogType
-from whylog.tests.tests_teacher import TestBase
+from whylog.constraints import TimeConstraint
+from whylog.tests.tests_teacher import TestBase, TestEffectBase, TestRuleBase
 
 from whylog.teacher.rule_validation_problems import (  # isort:skip
     NoEffectParserProblem, NotSetLogTypeProblem, NotUniqueParserNameProblem, ParserCountProblem
 )  # yapf: disable
+from whylog.constraints.validation_problems import (  # isort:skip
+    MinGreaterThatMaxProblem, ParamConversionProblem
+)
 
 
 class TestValidationBase(TestBase):
+    def _in_parser_problems(self, parser_id, problem):
+        validation_result = self.teacher.validate()
+        problems = validation_result.parser_problems.get(parser_id)
+        if problems is None:
+            return False
+        return problem in problems
+
     def _in_rule_problems(self, problem):
         validation_result = self.teacher.validate()
         return problem in validation_result.rule_problems
 
-    def _in_parser_problems(self, parser_id, problem):
+    def _in_constraint_problems(self, constraint_id, problem):
         validation_result = self.teacher.validate()
-        problems = validation_result.parser_problems.get(parser_id)
+        problems = validation_result.constraint_problems.get(constraint_id)
         if problems is None:
             return False
         return problem in problems
 
+
+class TestEffectValidationBase(TestEffectBase, TestValidationBase):
```

Comment:
I don't really know what going on here. You use multi inheriting. The programing rule `do not repeat yourself` not apply to tests. I guess it whould be much simpler and clearer if you just copy code instead of creating something like this.